lambdas/data-migration2/tests/test-pdrs.js

```diff
@@ -225,25 +235,116 @@ test.serial('migratePdrRecord handles nullable fields on source PDR data', async
   );
 });
 
-test.serial('migratePdrRecord throws RecordAlreadyMigrated error for already migrated record', async (t) => {
+test.serial('migratePdrRecord throws RecordAlreadyMigrated error if previously migrated record is newer', async (t) => {
   const { knex, testCollection, testProvider } = t.context;
 
-  const testPdr = generateTestPdr(testCollection, testProvider.name);
+  const testPdr = generateTestPdr({
+    collectionId: buildCollectionId(testCollection.name, testCollection.version),
+    provider: testProvider.name,
+  });
   await migratePdrRecord(testPdr, knex);
 
+  const olderTestPdr = {
+    ...testPdr,
+    updatedAt: Date.now() - 1000,
+  };
+
   await t.throwsAsync(
-    migratePdrRecord(testPdr, knex),
+    migratePdrRecord(olderTestPdr, knex),
     { instanceOf: RecordAlreadyMigrated }
   );
 });
 
+test.serial('migratePdrRecord throws RecordAlreadyMigrated error if the record was previously migrated and the new record is "running"', async (t) => {
+  const { knex, testCollection, testProvider } = t.context;
+
+  const testPdr = generateTestPdr({
+    collectionId: buildCollectionId(testCollection.name, testCollection.version),
+    provider: testProvider.name,
+  });
+  await migratePdrRecord(testPdr, knex);
+
+  const newerTestPdr = {
+    ...testPdr,
+    updatedAt: Date.now(),
+    status: 'running',
+  };
+
+  await t.throwsAsync(
+    migratePdrRecord(newerTestPdr, knex),
+    { instanceOf: RecordAlreadyMigrated }
+  );
+});
+
+test.serial('migratePdrRecord updates an already migrated record if the updated date is newer', async (t) => {
+  const {
+    knex,
+    testCollection,
+    testProvider,
+    pdrPgModel,
+    providerCumulusId,
+    collectionCumulusId,
+  } = t.context;
+
+  const executionPgModel = new ExecutionPgModel();
+  const execution = fakeExecutionRecordFactory();
+
+  const executionResponse = await executionPgModel.create(
+    knex,
+    execution
+  );
+  const executionCumulusId = executionResponse[0];
+
+  const testPdr = generateTestPdr({
+    collectionId: buildCollectionId(testCollection.name, testCollection.version),
+    provider: testProvider.name,
+    execution: execution.arn,
+    status: 'completed',
+  });
+  await migratePdrRecord(testPdr, knex);
+
+  const newerTestPdr = {
+    ...testPdr,
+    updatedAt: Date.now(),
+  };
+  console.log(testPdr.updatedAt);
+  console.log(newerTestPdr.updatedAt);
```

Comment:
Remove the console logs

Reply:
Ugh. Embarrassing.

https://github.com/nasa/cumulus/pull/2162/commits/cd3706e8988af761d806400b7603bef0dcb7fb37